cpp/src/arrow/array/builder_binary.h

```diff
@@ -136,70 +276,119 @@ class ARROW_EXPORT BinaryBuilder : public ArrayBuilder {
   /// Temporary access to a value.
   ///
   /// This pointer becomes invalid on the next modifying operation.
-  const uint8_t* GetValue(int64_t i, int32_t* out_length) const;
+  const uint8_t* GetValue(int64_t i, offset_type* out_length) const {
+    const offset_type* offsets = offsets_builder_.data();
+    const auto offset = offsets[i];
+    if (i == (length_ - 1)) {
+      *out_length = static_cast<offset_type>(value_data_builder_.length()) - offset;
+    } else {
+      *out_length = offsets[i + 1] - offset;
+    }
+    return value_data_builder_.data() + offset;
+  }
 
   /// Temporary access to a value.
   ///
   /// This view becomes invalid on the next modifying operation.
-  util::string_view GetView(int64_t i) const;
+  util::string_view GetView(int64_t i) const {
+    const offset_type* offsets = offsets_builder_.data();
```

Comment:
can this be delegated to GetValue(int64_t i, offset_type* out_length)?

Reply:
Good idea, yes.